lock/src/main/java/com/auth0/android/lock/CustomField.java

```diff
@@ -0,0 +1,159 @@
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import static com.auth0.android.lock.CustomField.FieldType.TYPE_COUNTRY;
+import static com.auth0.android.lock.CustomField.FieldType.TYPE_DATE;
+import static com.auth0.android.lock.CustomField.FieldType.TYPE_EMAIL;
+import static com.auth0.android.lock.CustomField.FieldType.TYPE_PERSON_NAME;
+import static com.auth0.android.lock.CustomField.FieldType.TYPE_PHONE_NUMBER;
+import static com.auth0.android.lock.CustomField.FieldType.TYPE_WORK_NAME;
+
+public class CustomField implements Parcelable {
+
+    @IntDef({TYPE_PERSON_NAME, TYPE_DATE, TYPE_WORK_NAME, TYPE_COUNTRY, TYPE_PHONE_NUMBER, TYPE_EMAIL})
+    @Retention(RetentionPolicy.SOURCE)
+    public @interface FieldType {
+        int TYPE_PERSON_NAME = 0;
+        int TYPE_DATE = 1;
+        int TYPE_WORK_NAME = 2;
+        int TYPE_COUNTRY = 3;
+        int TYPE_PHONE_NUMBER = 5;
+        int TYPE_EMAIL = 6;
+    }
+
+    private final String key;
+    private final String hint;
+    private final int type;
+
+    public CustomField(@FieldType int type, @NonNull String key, @NonNull String hint) {
+        if (hint.isEmpty()) {
+            throw new IllegalArgumentException("The hint cannot be empty!");
+        }
+        if (key.isEmpty()) {
+            throw new IllegalArgumentException("The key cannot be empty!");
+        }
+        this.key = key;
+        this.hint = hint;
+        this.type = type;
+
+    }
+
+    public void configureField(@NonNull ValidatedInputView field) {
+        switch (type) {
+            case TYPE_PERSON_NAME:
+                field.setDataType(ValidatedInputView.DataType.PERSON);
+                break;
+            case TYPE_DATE:
+                field.setDataType(ValidatedInputView.DataType.DATE);
+                break;
+            case TYPE_WORK_NAME:
+                field.setDataType(ValidatedInputView.DataType.WORK);
+                break;
+            case TYPE_COUNTRY:
+                field.setDataType(ValidatedInputView.DataType.COUNTRY);
+                break;
+            case TYPE_PHONE_NUMBER:
+                field.setDataType(ValidatedInputView.DataType.PHONE_NUMBER);
+                break;
+            case TYPE_EMAIL:
+                field.setDataType(ValidatedInputView.DataType.EMAIL);
+                break;
+        }
+        field.setHint(hint);
+        field.setTag(key);
+    }
+
+    @Nullable
+    public String findValue(@NonNull ViewGroup container) {
+        String value = null;
+        View view = container.findViewWithTag(key);
+        if (view != null) {
+            value = ((ValidatedInputView) view).getText();
+        }
+        return value;
+    }
+
+    public String getKey() {
+        return key;
+    }
+
+    protected CustomField(Parcel in) {
+        key = in.readString();
+        hint = in.readString();
+        type = in.readInt();
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
If you override this you are *bound* to override `hashCode()`. Why we need to override the `equals`?